test/probes/http-common.js

```diff
@@ -481,6 +481,36 @@ describe(`probes.${p}`, function () {
       })
     })
 
+    it('should map a specific header to ClientIP when specified', function (done) {
+      const server = createServer(options, function (req, res) {
+        res.end('done');
+      });
+
+      ao.probes[p]['client-ip-header'] = 'x-real-ip';
+      const ClientIPExpected = '777.777.333.333';
+
+      helper.doChecks(emitter, [
+        function (msg) {
+          check.server.entry(msg);
+          expect(msg).property('ClientIP', ClientIPExpected);
+        },
+        function (msg) {
+          check.server.exit(msg);
+        }
+      ], function () {
+        server.close(done);
+      });
+
+      server.listen(function () {
+        const port = server.address().port;
+        const options = {
+          url: `${p}://localhost:${port}`,
+          headers: {'x-real-ip': ClientIPExpected}
+        }
+        request(options);
+      });
+    })
+
```

Comment:
can you test that it works when the header is undefined?
I think that is the test case we want to make sure doesn't break with future code changes.

Reply:
i can't test what i really want to (how does the client handle not having a `socket.remoteAddress` property without spending a lot of time on the test. i'd have to patch the cached http/https modules to get control before the probe sees the `req` object or i'd have to figure out what's going on with nginx/nodejs that causes `remoteAddress` to be undefined - i'm not sure how the TCP protocol can fail to include a source IP address.

i added a specific test as a placeholder for a more sophisticated test in the future. currently all tests verify that ClientIP is present in the KV pairs when 'client-ip-header' is falsey.